egs/gale_arabic/s5/local/gale_prep_dict.sh

```diff
@@ -25,8 +25,7 @@ echo SIL > $dir/optional_silence.txt
 cat $dir/lexicon.txt | cut -d ' ' -f2- | tr -s ' ' '\n' |\
 sort -u >  $dir/nonsilence_phones.txt || exit 1;
 
-
- sed -i '1i<UNK> SIL' $dir/lexicon.txt
+ ex -s -c '1i|<UNK> SIL' -c x $dir/lexicon.txt
```

Comment:
This you can also do with perl alone
```
perl -i -pe 'print "<UNK> SIL\n" if $.==1'  $dir/lexicon.txt
```
Since the printing of the `-p` happens *after* all other processing (see `perldoc perlrun` for the exact loop perl runs with the `-p`), the line is inserted *before* the first (`$. == 1`) line in the file.